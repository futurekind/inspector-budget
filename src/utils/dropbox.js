const KEY = 'IB_AT';
const APP_KEY = '4inxggnvthsk657';
const AUTHORIZE_URL = 'https://www.dropbox.com/1/oauth2/authorize'

export const shouldGetAccessToken = () => {
    if(localStorage.getItem(KEY)) {
        return false
    } else {
        return true
    }
}

export const authorize = () => {
    window.location.href = `${AUTHORIZE_URL}?response_type=token&client_id=${APP_KEY}&redirect_uri=http://localhost:3000/`
}

export const getAccessTokenInHashFragment = hashStr => {
    const fragments = hashStr.split('&').reduce((comp, fragment) => {
        const piece = fragment.split('=');
        return Object.assign({}, comp, {
            [piece[0]]:piece[1]
        });
    }, {});
    
    return fragments.access_token || false;
}

export const setAccessToken = token => {
    localStorage.setItem(KEY, token);
}

export const getAccessToken = () => localStorage.getItem(KEY);