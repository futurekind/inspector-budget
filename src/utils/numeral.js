import numeral from 'numeral';
import de from 'numeral/locales/de';

numeral.defaultFormat('0,0.00')
numeral.locale('de');

export { numeral };