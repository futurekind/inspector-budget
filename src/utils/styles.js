export const colors = {
    black: '#000',
    dark: '#333',
    light: '#888'
}

export const sizes = {
    navigationWidth: 90
}