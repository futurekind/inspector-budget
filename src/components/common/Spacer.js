import React, { PropTypes } from 'react';
import { sizes } from '../../utils/styles';

const getStyle = props => {
    switch(props.type) {

        case 'top':
            return {
                paddingTop: props.value * sizes.spacer,
            }

        case 'bottom':
            return {
                paddingBottom: props.value * sizes.spacer
            }
        
        case 'topbottom':
        default:
            return {
                paddingTop: props.value * sizes.spacer,
                paddingBottom: props.value * sizes.spacer
            }
    }
}

const Spacer = (props) => {
    return (
        <div style={ getStyle(props) }>
            { props.children }
        </div>
    )
}

Spacer.defaultProps = {
    type: 'topbottom',
    value: 1
}

Spacer.propTypes = {
    type: PropTypes.oneOf([
        'topbottom', 'top', 'bottom'
    ]),
    value: PropTypes.oneOf([
        0, 1, 2, 3, 4, 5, 6, 7, 8
    ])
}

Spacer.displayName =  'Spacer';

export default Spacer