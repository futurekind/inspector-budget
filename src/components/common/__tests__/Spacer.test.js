import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { sizes } from '../../../utils/styles'
import Spacer from '../Spacer';

describe('Spacer', () => {
    const wrapper = shallow(<Spacer />);
    
    it('renders as excepted', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders its children', () => {
        const wrapper = shallow(<Spacer><p>some text</p></Spacer>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders just top space when type prop is set', () => {
        wrapper.setProps({
            type: 'top'
        })
        const style = wrapper.prop('style');   

        expect(style.paddingBottom).not.toBeDefined();
    })

    it('renders just bottom space when type prop is set', () => {
        wrapper.setProps({
            type: 'bottom'
        })
        const style = wrapper.prop('style');   
        
        expect(style.paddingTop).not.toBeDefined();
    })
    
    it('calculates the paddingTop and paddingBottom values correctly', () => {
        wrapper.setProps({
            value: 2,
            type: 'topbottom'
        })
        const style = wrapper.prop('style');

        expect(style.paddingTop).toBe(sizes.spacer * 2)
        expect(style.paddingBottom).toBe(sizes.spacer * 2)
    })

    it('calculates the paddingTop value correctly', () => {
        wrapper.setProps({
            value: 8,
            type: 'top'
        })
        const style = wrapper.prop('style');

        expect(style.paddingTop).toBe(sizes.spacer * 8)
    })

    it('calculates the paddingBottom value correctly', () => {
        wrapper.setProps({
            value: 3,
            type: 'bottom'
        })
        const style = wrapper.prop('style');

        expect(style.paddingBottom).toBe(sizes.spacer * 3)
    })

})