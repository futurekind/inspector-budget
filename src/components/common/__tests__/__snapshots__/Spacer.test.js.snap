exports[`Spacer renders as excepted 1`] = `
<div
  style={
    Object {
      "paddingBottom": 12,
      "paddingTop": 12,
    }
  } />
`;

exports[`Spacer renders its children 1`] = `
<div
  style={
    Object {
      "paddingBottom": 12,
      "paddingTop": 12,
    }
  }>
  <p>
    some text
  </p>
</div>
`;
